nova/virt/vmwareapi/volumeops.py

```diff
@@ -73,14 +84,25 @@ def attach_disk_to_vm(self, vm_ref, instance,
                    'device_name': device_name, 'disk_type': disk_type},
                   instance=instance)
 
-    def _update_volume_details(self, vm_ref, volume_uuid, device_uuid):
-        # Store the uuid of the volume_device
+    def _update_volume_details(self, vm_ref, volume_uuid, device_uuid,
+                               config_spec=None):
+        """Store the UUID of the volume's device in extraConfig
+
+        If "config_spec" is given, updates the config_spec only instead of
```

Comment:
I would prefer to see two different functions: The old `_update_volume_details`, which does reconfigure the vm,
and a new one, which only updates the config-spec (which then can be used by the old function), and depending on what should happen, the appropriate function should be called.

Reply:
Ok. I'll rename the function to `_add_volume_details_to_config_spec()` and delete the reconfigure. There is no other caller.